NewForce-Cohort-7/tabloidmvc-lone-star-cowboys
# View comments by post

Description
This pull request is for TabloidMVC. 

Trello ticket #19 - View a Post's Comments,  https://trello.com/c/KrqAoADd/19-19-view-a-posts-comments. 

The branch for the pull request: sl-branch1

A user can is now able to go to post details and see a link named "View all Comments." If there are comments, the user will see a list of comments for that specific post. They will see subject, content, datecreated, and name of author. Additionally the user can go return to the previous page by clicking the link "back to posts"

Updates:
- Created files necessary to make comments: Comment.cs, CommentController.cs, CommentRepository.cs, ICommentRepository.cs
- Updated Program.cs to recognize ICommentRepository and CommentRepository
-  Added List Methods GetAllComments() and GetCommentByPostId() as well as the GetCommentById() method
- Added above methods to ICommentRepository
- Added a View called Index.cshtml For Comment views folder
- Added View Comments to details.cshtml for Post in order to access comments through the post details

Type of change:
[ x ] New feature (non-breaking change which adds functionality)

Testing Instructions:
Follow these instructions to protect the branch you are working on and to test this pull request:  https://gist.github.com/jordan-castelloe/96f76d5bd4cfbd2181084af674c2a1af#when-you-need-to-test-a-teammates-pr
If you do not currently have the code for this project, clone the repository -  https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys.
Open the TabloidMVC.sln in visual studio
	• In the github repository:  https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys
	• Open the SQL folder:  https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/tree/main/SQL
	• there are two files - [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/blob/main/SQL/01_TabloidMVC_Create_DB.sql) and [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)
	• In Visual Studio in the top navbar click View and select SQL Server Object Explorer. Open the server marked as "SQLEXPRESS" in the SQL Server Object Explorer on the left side of Visual Code. Once there, add a new SQL server by selecting the 'Add SQL Server' option at the top of the side bar. Copy the data from the [seed data](https://github.com/NewForce-Cohort-7/tabloidmvc-lone-star-cowboys/tree/main/SQL). Prepare your database. In order to see comments in the database, open a new SQL Query and seed the database with this additional data:

	INSERT INTO [Comment] (PostId, UserProfileId, Subject, Content, CreateDateTime) VALUES (1, 1, 'Technology', 'I love this post!', '2020-04-08 16:00:00')
	SELECT * FROM Comment
	(This will manually inserts a comment into the database since create hasn't been executed yet.)

	• execute by pressing the green Play button.
	• start the server by clicking the solid green play button in the middle top of your screen. A browser window will display with the Tabloid landing page.
	• In the nav bar, click Login. You only need an email to login use [admin@example.com](mailto:admin@example.com)
	• You now see the home page for the site. In the navbar, click on "Posts"
You will see the posts page with this specific post: "C# is the Best Language admin Technology 6/29/2023."
	1. Click on the eyeball icon. This will take you to post details
	2. Scroll to the bottom and click "View Comments"
	3. You should now a table that has one comment in it. There is a subject, content, Date Created, and Author name. Do not click the "Create", "Edit", "Details", or "Delete" links. These features have not been implemented yet.
	4. You may now click "Back to Post to return to post details. Then click "Back to Posts" again to return to Posts.
	5. 
If you would like to further test this or test it through a SQL query, create a query for the database you created at the start of the pull request testing. In your new query, paste:
SELECT * FROM Comment

After running the query (the green play button on the left of the query window) you should see all the data associated for the comment that you just saw in the browser.
